src/hooks/room/useRoomMemberProfile.ts

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { RoomMember } from "matrix-js-sdk/src/models/room-member";
+import { useContext, useEffect, useState } from "react";
+
+import RoomContext, { TimelineRenderingType } from "../../contexts/RoomContext";
+import { useSettingValue } from "../useSettings";
+
+export function useRoomMemberProfile({
+    userId = "",
+    member: propMember,
+    forceHistorical = false,
+}: {
+    userId: string | undefined;
+    member?: RoomMember;
+    forceHistorical?: boolean;
+}): RoomMember {
+    const [member, setMember] = useState<RoomMember | undefined>(propMember);
+
+    const context = useContext(RoomContext);
```

Comment:
You can also use `useRoomContext()` here but just a question of preference :)

Reply:
Uhh TIL !

Reply:
I added it few weeks ago ^^